### PR TITLE
fix: allow to delete window[`on${event}`] property

### PR DESCRIPTION
**Pull request by Fetz**
Hi here is the possible fix for the issue jsdom/jsdom#2140

- the to-upstream test file window-properties.html that I created is basically a duplicated of:
w3c web-platform-tests [the-window-object/window-properties.html](https://github.com/w3c/web-platform-tests/blob/master/html/browsers/the-window-object/window-properties.html), but with the other breaking tests commented out.
- added some missing events like: "mousewheel", "show"
- added enumerable to events created with create-event-accessor.js
- for some reason, even if we are adding getter/setter to the {global|window}Events of the window, when the window is being used in the tests the getter/setter doesn't show in the test for some reason. I will probably will need help to fix it if is a blocker

Let me know if there is something I need to tweak or improve

### CODE REVIEW

File: lib/jsdom/living/nodes/GlobalEventHandlers-impl.js [hunk 1]
```diff
@@ -22,14 +22,14 @@ const events = new Set([
...
-  "mouseup", "wheel",
+  "mouseup", "wheel", "mousewheel",
   "pause", "play",
   "playing", "progress",
   "ratechange", "reset",
   "resize", "scroll",
   "securitypolicyviolation",
   "seeked", "seeking",
-  "select", "sort", "stalled",
+  "select", "show", "sort", "stalled",
```
Comment (TimothyGu) [inline lib/jsdom/living/nodes/GlobalEventHandlers-impl.js]: These two additions don't seem to be in the [spec](https://html.spec.whatwg.org/multipage/webappapis.html#globaleventhandlers) anymore. See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/GlobalEventHandlers.onmousewheel) for `mousewheel`.

**Comment by Fetz** [comment]
The 3 tests that are failing, in my machine they are working. I suppose is because I'm using a different version of node (v8.5.0).

There are some linting warning in the tests, should I fix even if they will make it different from the source (like 'prefer arrow callback')

File: lib/jsdom/living/nodes/GlobalEventHandlers-impl.js (hunk 1 above)
Comment (Fetz) [reply lib/jsdom/living/nodes/GlobalEventHandlers-impl.js]: but they are still on the test from https://github.com/w3c/web-platform-tests/blob/master/html/browsers/the-window-object/window-properties.html, should I ignore for our case and probably later do a contribution there to remove these 2 additions?

Comment (TimothyGu) [reply lib/jsdom/living/nodes/GlobalEventHandlers-impl.js]: Yeah that works.

**Comment by Fetz** [comment]
Yes, I will rename the file later 👍 , regarding the tests I will see if there is something I can do (since looking at the build log looks like the tests that are failing are the ones that test the primitives values).

If is a problem with node v.6 I will remove or comment those tests (depending of what is the best for this project)

**Comment by TimothyGu** [comment]
> There are some linting warning in the tests, should I fix even if they will make it different from the source (like 'prefer arrow callback')

Yes.

**Comment by TimothyGu** [comment]
> If is a problem with node v.6 I will remove or comment those tests (depending of what is the best for this project)

The same failures [appear](https://travis-ci.org/jsdom/jsdom/jobs/336826288#L5679) with Node.js v9 as well so I'm not sure if it's just a version issue.

**Comment by Zirro** [comment]
Great work, thanks for helping us debug the issue as well as fixing it!

> There are some linting warning in the tests, should I fix even if they will make it different from the source (like 'prefer arrow callback')
>> Yes

Tests which include `dont-upstream` in their name get [ignored](https://github.com/jsdom/jsdom/blob/master/.eslintignore#L30) by the linter now, so I think it'd be better to just rename the file without introducing differences to satisfy it.

**Comment by Fetz** [comment]
the test failing looks like an issue with node, I created an issue in [nodejs project](https://github.com/nodejs/node/issues/18551). So I probably will comment those failing tests.

Regarding those failing tests is ok to be comment or is better to delete them?

**Comment by TimothyGu** [comment]
@Fetz :disappointed: feel free to comment them out then. I'll take a look at them some time in the future as part of the Window overhaul.

**Comment by Fetz** [comment]
The tests are now passing, I removed as well those 2 events handlers and renamed the file. Let me know if I need to do something else

File: test/web-platform-tests/to-upstream/html/browsers/the-window-object/window-properties-dont-upstream.html [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+  "atob",
+
+  // WindowTimers
+  "setTimeout",
+  "clearTimeout",
+  "setInterval",
+  "clearInterval",
+
+  // HTML Editing APIs
+  // "getSelection", //doesn't work for some reason
```
Comment (Zirro) [inline test/web-platform-tests/to-upstream/html/browsers/the-window-object/window-properties-dont-upstream.html]: The reason that these methods/attributes - `getSelection`, `matchMedia`, `applicationCache`, `sessionStorage` and `localStorage` - don't work is that they have not yet been implemented. Would be good if you could change the comments to clarify that.

With that, LGTM.

**Pull request merged**